### PR TITLE
Update sum_of_digits.py

### PR DESCRIPTION
**Pull request by shellhub**
### **Describe your change:**

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/sum_of_digits.py [hunk 1]
```diff
@@ -6,13 +6,41 @@ def sum_of_digits(n: int) -> int:
...
+
+def main():
+    import doctest
+
+    doctest.testmod()
+
+
 if __name__ == "__main__":
-    print(sum_of_digits(12345))  # ===> 15
+    main()
```
Comment (cclauss) [inline maths/sum_of_digits.py:46]: Why remove this line?

Comment (shellhub) [reply maths/sum_of_digits.py:46]: Already test using doctest

**Comment by cclauss** [comment]
def sum_of_digits_compact(n: int) -> int:
    return sum(int(c) for c in str(n).lstrip("-"))

**Comment by shellhub** [comment]
> ```python
> def sum_of_digits_compact(n: int) -> int:
>     return sum(int(c) for c in str(n).lstrip("-"))
> ```

Thanks.

**Comment by shellhub** [comment]
Please review again. I have added benchmark @cclauss

**Pull request merged**